packages/client/src/components/reports/scape/ScapeExecutionsInProgressResult.tsx

```diff
@@ -0,0 +1,250 @@
+  selectedCanaryExecutionResult.executionStatus === stageStatus.TERMINAL && selectedCanaryExecutionResult.exception;
+
+@observer
+export default class ScapeExecutionsInProgressResult extends React.Component<Props, State> {
+  constructor(props: Readonly<Props>) {
+    super(props);
+
+    this.state = {
+      selectedCAEIndex: safeGet(() => this.props.results.canaryExecutionResults.length - 1).orElse(0),
+      isSelectedRunComplete: safeGet(() => this.props.results.canaryExecutionResults.length > 0).orElse(false)
+    };
+  }
+
+  onCAETabClick(index: number, status: string) {
+    this.setState({ selectedCAEIndex: index, isSelectedRunComplete: status !== stageStatus.IN_PROGRESS });
+  }
+
+  render(): React.ReactNode {
+    const {
+      stageStatusList,
+      application,
+      user,
+      metricSourceType,
+      applicationMetadata,
+      metricsAccountName,
+      storageAccountName,
+      startTime,
+      lifetime,
+      thresholds,
+      results,
+      selectedCanaryExecutionResult,
+      result,
+      selectedMetric,
+      request,
+      canaryConfig,
+      canaryAnalysisResultByIdMap,
+      idListByMetricGroupNameMap,
+      groupScoreByMetricGroupNameMap,
+      metricSetPairsByIdMap,
+      classificationCountMap,
+      metricGroupNamesDescByWeight,
+      displayMetricOverview,
+      handleOverviewSelection,
+      handleCanaryRunSelection,
+      handleMetricSelection,
+      handleGoToConfigButtonClick
+    } = this.props;
+
+    const selectedRunNumber = this.state.selectedCAEIndex + 1;
+    const runCanaryStages = stageStatusList.filter(stageMetadata => stageMetadata.type === 'runCanary');
+
+    const determineRefereeStageMetadata = (stageMetadata: StageMetadata, index: number): RefereeStageMetadata => {
+      let score: string = '...';
+      let status: string = 'in-progress';
+      let availableResult: CanaryExecutionResult | undefined = undefined;
+
+      if (stageMetadata.status === stageStatus.TERMINAL) {
+        score = '0';
+        status = 'fail';
+      } else if (stageMetadata.status === stageStatus.SUCCEEDED) {
+        safeGet(() => results.canaryExecutionResults).ifPresent(
+          (results: CanaryExecutionResult[]) =>
+            (availableResult = results.find(result => result.executionId === stageMetadata.executionId))
+        );
+        if (availableResult) {
+          score = safeGet(() => availableResult)
+            .get()
+            .result.judgeResult!.score.score.toFixed(0)
+            .toString();
+          status = ScoreClassUtils.getClassFromScore(parseInt(score), results.canaryScores, thresholds, index);
+        }
+      }
+
+      return new RefereeStageMetadata(score, status, availableResult);
+    };
+
+    return (
+      <div className="scape-in-progress-report-container">
+        <div className="scape-metadata-container">
+          <TitledSection title="Canary Report" />
+          <ScapeMetadataSection
```

Comment:
Consider using https://reactjs.org/docs/jsx-in-depth.html#spread-attributes to pass props to this component.